SplitTests/Storage/ImpressionsStorageTest.swift

```diff
@@ -0,0 +1,96 @@
+//
+//  ImpressionsStorageTest.swift
+//  SplitTests
+//
+//  Created by Javier Avrudsky on 25-Nov-2022.
+//  Copyright © 2022 Split. All rights reserved.
+//
+
+import Foundation
+
+import XCTest
+@testable import Split
+
+class ImpressionsStorageTest: XCTestCase {
+
+    var persistentStorage: PersistentImpressionsStorageStub!
+
+    override func setUp() {
+        persistentStorage = PersistentImpressionsStorageStub()
+    }
+
+//    func enablePersistence(_ enable: Bool)
+//    func push(_ impression: KeyImpression)
+//    func clearInMemory()
+
+    func testStartDisabledPersistence() {
+        let impressionsStorage = MainImpressionsStorage(persistentStorage: persistentStorage,
+                                                        persistenceEnabled: false)
+
+        TestingHelper.createKeyImpressions(feature: "f1", count: 10).forEach {
+            impressionsStorage.push($0)
+        }
+
+        let count = persistentStorage.storedImpressions.count
+
+        XCTAssertEqual(0, count)
+    }
+
+    func testStartEnabledPersistence() {
+        let impressionsStorage = MainImpressionsStorage(persistentStorage: persistentStorage,
+                                                        persistenceEnabled: true)
+
+        TestingHelper.createKeyImpressions(feature: "f1", count: 10).forEach {
+            impressionsStorage.push($0)
+        }
+
+        let count = persistentStorage.storedImpressions.count
+
+        XCTAssertEqual(10, count)
+    }
+
+    func testEnablePersistence() {
```

Comment:
Maybe add `testDisablePersistence`? if it makes sense

Reply:
👍